controllers/api/posts.js

```diff
@@ -81,4 +83,60 @@ async function destroy(req, res, next) {
     } catch (error) {
         res.status(400).json({ msg: error.message })
     }
-}
\ No newline at end of file
+}
+
+// Like Post
+async function likePost(req, res, next) {
+    try {
+        const postId = req.params.id
+        const post = await Post.findById(postId)
+
+        if (!post) {
+            return res.status(404).json({ message: 'Post not found' })
+        }
+
+        // Add post to likedPosts array of the user if not already present
+        if (!req.user.likedPosts.includes(post._id)) {
+            req.user.likedPosts.push(post._id)
+            await req.user.save()
+        }
+
+        // Add user to likes array of the post if not already present
+        if (!post.likes.includes(req.user._id)) {
+            post.likes.push(req.user._id)
+            await post.save()
+        }
+
+        res.locals.data.post = post
+        next()
+    } catch (error) {
+        console.error("Error liking post:", error)
+        res.status(500).json({ message: "Internal server error" })
+    }
+}
+
+// Unlike Post
+async function unlikePost(req, res, next) {
+    try {
+        const postId = req.params.id
+        const post = await Post.findById(postId)
+
+        if (!post) {
+            return res.status(404).json({ message: 'Post not found' })
+        }
+
+        // Remove post from likedPosts array of the user
+        req.user.likedPosts = req.user.likedPosts.filter(likedPostId => likedPostId.toString() !== post._id.toString())
```

Comment:
modify this to throw an error if `user.likedPosts` doesn't contain the `post._id`